fbw-a32nx/src/systems/fmgc/src/components/fms-messages/TuneNavaid.ts

```diff
@@ -0,0 +1,51 @@
+import { NavaidTuner } from '@fmgc/navigation/NavaidTuner';
+import { FMMessage, FMMessageTypes } from '@shared/FmMessages';
+import { Trigger } from '@shared/logic';
+import { FMMessageSelector, FMMessageUpdate } from './FmsMessages';
+
+abstract class TuneNavaid implements FMMessageSelector {
+    message: FMMessage = ({ ...FMMessageTypes.TuneNavaid });
+
+    abstract efisSide: 'L' | 'R';
+
+    private trigRising = new Trigger(true);
+
+    private trigFalling = new Trigger(true);
+
+    private navaidTuner: NavaidTuner;
+
+    init(baseInstrument: BaseInstrument): void {
+        this.navaidTuner = baseInstrument.navigation.getNavaidTuner();
+    }
+
+    process(deltaTime: number): FMMessageUpdate {
+        const message = this.navaidTuner.getTuneNavaidMessage();
+
+        this.trigRising.input = message !== null;
+        this.trigRising.update(deltaTime);
+
+        this.trigFalling.input = message === null;
+        this.trigFalling.update(deltaTime);
```

Comment:
Doesn't this result in ``update(deltaTime)`` being called twice per tick? Is this intentional?

Reply:
Nah, two different triggers. Oddly the falling one is actually created as a rising edge trigger and the input inverted. I guess there was a bug with the falling edge trigger when we did the first message and it's been copy-pasted to the others ever since.